driver/labjack/writer.h

```diff
@@ -0,0 +1,222 @@
+    synnax::Rate state_rate = synnax::Rate(1);
+    std::map<synnax::ChannelKey, labjack::out_state> state_map;
+    synnax::ChannelKey state_index_key;
+    loop::Timer timer;
+}; // class StateSource
+
+///////////////////////////////////////////////////////////////////////////////////
+//                                   WriterChannelConfig                         //
+///////////////////////////////////////////////////////////////////////////////////
+struct WriterChannelConfig {
+    std::string location;
+    bool enabled = true;
+    synnax::DataType data_type;
+    uint32_t cmd_key;
+    uint32_t state_key;
+    std::string channel_type = "";
+
+    WriterChannelConfig() = default;
+
+    explicit WriterChannelConfig(config::Parser &parser)
+        : enabled(parser.optional<bool>("enabled", true)),
+          data_type(parser.optional<std::string>("data_type", "uint8")),
+          cmd_key(parser.required<uint32_t>("cmd_key")),
+          state_key(parser.required<uint32_t>("state_key")),
+          channel_type(parser.optional<std::string>("channel_type", "")),
+          location(parser.optional<std::string>("port", "")) {
+        if (!parser.ok())
+            LOG(ERROR) << "Failed to parse writer channel config: " << parser.
+                    error_json().dump(4);
+    }
+};
+
+///////////////////////////////////////////////////////////////////////////////////
+//                                   WriterConfig                                //
+///////////////////////////////////////////////////////////////////////////////////
+struct WriterConfig {
+    std::string device_type;
+    std::string device_key;
+    std::vector<WriterChannelConfig> channels;
+    synnax::Rate state_rate = synnax::Rate(1);
+    std::string serial_number; // used to open devices
+    std::string connection_type;
+    bool data_saving;
+    std::string task_name;
+    synnax::ChannelKey task_key;
+    std::map<synnax::ChannelKey, labjack::out_state> initial_state_map;
+    synnax::ChannelKey state_index_key;
+
+    WriterConfig() = default;
+
+    explicit WriterConfig(config::Parser &parser)
+        : device_type(parser.optional<std::string>("type", "")),
+          device_key(parser.required<std::string>("device")),
+          state_rate(synnax::Rate(parser.optional<int>("state_rate", 1))),
+          serial_number(parser.required<std::string>("device")),
+          connection_type(parser.optional<std::string>("connection_type", "")),
+          data_saving(parser.optional<bool>("data_saving", false)
+          ) {
+        if (!parser.ok())
+            LOG(ERROR) << "Failed to parse writer config: " << parser.error_json().
+                    dump(4);
+        // Parse the channels
+        parser.iter("channels", [this](config::Parser &channel_parser) {
+            channels.emplace_back(WriterChannelConfig(channel_parser));
+
+            double initial_val = 0.0;
+            if (channels.back().data_type == synnax::SY_UINT8) {
```

Comment:
This is unintuitive. Can you add a comment explaining why this should be a 1 instead of a 0?